src/debugpy/server/tracing/__init__.py

```diff
@@ -255,21 +272,24 @@ def stack_trace_len(self) -> int:
         """
         Returns the total count of frames in this thread's stack.
         """
-        try:
-            with _cvar:
-                python_frame = self.current_frame
-        except ValueError:
-            raise ValueError(f"Can't get frames for inactive Thread({self.id})")
-        try:
-            return len(tuple(traceback.walk_stack(python_frame)))
-        finally:
-            del python_frame
+        return len(self._get_stack_trace())
 
     def stack_trace(self) -> Iterable["StackFrame"]:
         """
         Returns an iterable of StackFrame objects for the current stack of this thread,
         starting with the topmost frame.
         """
+        return self._get_stack_trace()
+    
+    def _get_stack_trace(self) -> list["StackFrame"]:
+        # If our current frame is none, this is invalid. Throw an error.
+        if self._current_frame is None:
+            raise InvalidMessageError(reason="Tread is not suspended")
```

Comment:
This will crash the handler since `InvalidMessageError` will not be associated with the request, and so it won't be treated as a response to it automatically. It should use some non-messaging-specific exception type - I think `ValueError` might be appropriate here - and then the handler should catch it and do `return request.isnt_valid(...)`. Or the handler can check `current_frame` itself first, before it even tries to call `stack_trace` and `stack_trace_len`.

BTW, I don't recall if `stack_trace` and `stack_trace_len` were called anywhere other than that one place in `stackTrace_request`. If they aren't, given that they are one-liners now, might as well inline them there.

Reply:
Okay I'll move the error check to the stackTrace_request. You were right, there's only one spot where this was called